packages/slice-machine/src/features/inAppGuide/InAppGuideContext.tsx

```diff
@@ -21,14 +22,18 @@ export const InAppGuideContext = createContext<
 
 export const InAppGuideProvider: FC<PropsWithChildren> = (props) => {
   const { children } = props;
+  const { eligible: isNewOnboardingEnabled } = useOnboardingExperiment();
   const [isInAppGuideOpen, setIsInAppGuideOpen] = usePersistedState(
     "isInAppGuideOpen",
     true,
   );
 
   const memoizedValue = useMemo(
-    () => ({ isInAppGuideOpen, setIsInAppGuideOpen }),
-    [isInAppGuideOpen, setIsInAppGuideOpen],
+    () => ({
+      isInAppGuideOpen: isNewOnboardingEnabled ? false : isInAppGuideOpen,
+      setIsInAppGuideOpen,
+    }),
+    [isInAppGuideOpen, setIsInAppGuideOpen, isNewOnboardingEnabled],
```

Comment:
We kinda do, because we're using `useInAppGuide` in another place, so the provider is above most of the component tree. This way the context still works but shows the value of `isInAppGuideOpen` is always off.

Reply:
in that case, i guess what we don't need is the other code in this PR, as the value will always be false, right?

Reply:
If we disabled the new onboarding flag, we want everything to work as it did before, so we should keep that code, no?

Reply:
yes, i mean that i believe we don't need to do the check twice

if we do it in the provider, i think that's sufficient and the other part where you check the flag is not needed, because that dialog is driven by this provider anyway (less code to touch and maintain)

Reply:
But what if a user has an open inAppGuide dialog? If we turn on the new onboarding, the `useInAppGuide` will return the value of the `isInAppGuideOpen` storage value, and we want to ignore it

Reply:
sorry, i'm confused, don't undertand what you're saying

my suggestion: leave this code, remove the flag check in the component, as it's anyway driven by the provider

Reply:
Clarified via DM. Done ✅